src/NUnitFramework/tests/Attributes/TestCaseSourceTests.cs

```diff
@@ -294,17 +294,10 @@ public void HandlesExceptionInTestCaseSource_GuiDisplay(string lhs, string rhs)
         private static IEnumerable<TestCaseData> ZeroTestCasesSource() => Enumerable.Empty<TestCaseData>();
 
         [TestCaseSource("ZeroTestCasesSource")]
-        public void TestWithZeroTestSourceCasesShouldNotBeRun()
-        {
-            Assert.Fail();
-        }
```

Comment:
Actually, no... This test, if left in, would now cause the run to fail since we are treating TestCaseSourceAttribute on a method without args as an error. Since it's an error, it can only be tested at second level, so we have TestMethodBuilderTests.TestCaseSourceAttribute_NoArgs_NoData_NotRunnable.

We could stand a great deal of refactoring of these tests for clarity, I'll admit.